DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -151,13 +94,10 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
                 throw new DeviceRequiredPropertyNotFoundException("'DeviceID' property is missing");
             }
 
-            string deviceId = device.DeviceProperties.DeviceID;
-
-            DeviceModel existingDevice = await this.GetDeviceAsync(deviceId);
-
+            DeviceModel existingDevice = await GetDeviceAsync(device.id);
```

Comment:
I've seen this method: GetDeviceAsync(device.id);  getting called with device.id and device.DeviceProperties.DeviceID. Which one is the correct one?